aws/resource_aws_route53_resolver_firewall_rule_group_association.go

```diff
@@ -0,0 +1,217 @@
+				ValidateFunc: validation.StringInSlice(route53resolver.MutationProtectionStatus_Values(), false),
+			},
+
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"vpc_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateRoute53ResolverName,
+			},
+
+			"tags":     tagsSchema(),
+			"tags_all": tagsSchemaComputed(),
+		},
+
+		CustomizeDiff: SetTagsDiff,
+	}
+}
+
+func resourceAwsRoute53ResolverFirewallRuleGroupAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
+
+	input := &route53resolver.AssociateFirewallRuleGroupInput{
+		CreatorRequestId:    aws.String(resource.PrefixedUniqueId("tf-r53-rslvr-frgassoc-")),
+		Name:                aws.String(d.Get("name").(string)),
+		FirewallRuleGroupId: aws.String(d.Get("firewall_rule_group_id").(string)),
+		Priority:            aws.Int64(int64(d.Get("priority").(int))),
+		VpcId:               aws.String(d.Get("vpc_id").(string)),
+		Tags:                tags.IgnoreAws().Route53resolverTags(),
+	}
+
+	if v, ok := d.GetOk("mutation_protection"); ok {
+		input.MutationProtection = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Creating Route 53 Resolver DNS Firewall rule group association: %#v", input)
+	output, err := conn.AssociateFirewallRuleGroup(input)
+	if err != nil {
+		return fmt.Errorf("error creating Route 53 Resolver DNS Firewall rule group association: %w", err)
+	}
+
+	d.SetId(aws.StringValue(output.FirewallRuleGroupAssociation.Id))
+
+	_, err = waiter.FirewallRuleGroupAssociationCreated(conn, d.Id())
+
+	if err != nil {
+		return fmt.Errorf("error waiting for Route53 Resolver DNS Firewall rule group association (%s) to become available: %w", d.Id(), err)
+	}
+
+	return resourceAwsRoute53ResolverFirewallRuleGroupAssociationRead(d, meta)
+}
+
+func resourceAwsRoute53ResolverFirewallRuleGroupAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	ruleGroupAssociation, err := finder.FirewallRuleGroupAssociationByID(conn, d.Id())
+
+	if isAWSErr(err, route53resolver.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] Route53 Resolver DNS Firewall rule group association (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error getting Route 53 Resolver DNS Firewall rule group association (%s): %w", d.Id(), err)
+	}
+
+	if ruleGroupAssociation == nil {
+		log.Printf("[WARN] Route 53 Resolver DNS Firewall rule group association (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
```

Comment:
Reference: https://github.com/hashicorp/terraform-provider-aws/issues/16796

```suggestion
	if ruleGroupAssociation == nil {
		if d.IsNewResource() {
			return fmt.Errorf("error getting Route 53 Resolver DNS Firewall rule group association (%s): not found after creation", d.Id())
		}

		log.Printf("[WARN] Route 53 Resolver DNS Firewall rule group association (%s) not found, removing from state", d.Id())
		d.SetId("")
		return nil
	}
```